spec/index.bs

```diff
@@ -11027,57 +11096,68 @@ GPUQueue includes GPUObjectBase;
+                            |dataBytes|.[=byte sequence/length=],
+                            |aspectSpecificFormat|,
+                            |size|) succeeds.
 
-                        <div class=validusage>
-                            - [$validating GPUImageCopyTexture$](|destination|, |size|) returns `true`.
-                            - |texture|.{{GPUTexture/usage}} includes {{GPUTextureUsage/COPY_DST}}.
-                            - |texture|.{{GPUTexture/sampleCount}} is 1.
-                            - [=validating texture copy range=](|destination|, |size|) return `true`.
-                            - |destination|.{{GPUImageCopyTexture/aspect}} must refer to a single aspect of
-                                |texture|.{{GPUTexture/format}}.
-                            - That aspect must be a valid image copy destination according to [[#depth-formats]].
-                            - Let |aspectSpecificFormat| = |texture|.{{GPUTexture/format}}.
-                            - If |texture|.{{GPUTexture/format}} is a [=depth-or-stencil format=]:
-                                - Set |aspectSpecificFormat| to the [=aspect-specific format=] of |texture|.{{GPUTexture/format}} according to [[#depth-formats]].
-                            - [$validating linear texture data$](|dataLayout|,
-                                |dataByteSize|,
-                                |aspectSpecificFormat|,
-                                |size|) succeeds.
-
-                            Note: unlike
-                            {{GPUCommandEncoder}}.{{GPUCommandEncoder/copyBufferToTexture()}},
-                            there is no alignment requirement on either
-                            |dataLayout|.{{GPUImageDataLayout/bytesPerRow}} or |dataLayout|.{{GPUImageDataLayout/offset}}.
-                        </div>
-                    1. Write |contents| into |destination|.
+                        Note: unlike
+                        {{GPUCommandEncoder}}.{{GPUCommandEncoder/copyBufferToTexture()}},
+                        there is no alignment requirement on either
+                        |dataLayout|.{{GPUImageDataLayout/bytesPerRow}} or |dataLayout|.{{GPUImageDataLayout/offset}}.
+                    </div>
+                1. Let |contents| be the contents of the [=images=] seen by
+                    viewing |dataBytes| with |dataLayout| and |size|.
 
-                        Issue: Specify more formally.
-                </div>
+                    Issue: Specify more formally.
+
+                    Note: This is described as copying all of |data| to the device timeline,
+                    but in practice |data| could be much larger than necessary.
+                    Implementations should optimize by copying only the necessary bytes.
+                1. Issue the subsequent steps.
+            </div>
+            <div data-timeline=queue>
+                On the [=Queue timeline=] of |this|:
```

Comment:
nit: The only thing that feels slightly strange to me (and it's not a huge deal) is that "Issue the subsequent steps" and "On the ___ timeline of |whatever|" feels like a single sentence that's been split across the two sections.

It would seem to me like we could say "1. Issue the subsequent steps on the ____ timeline of |whatever|." at the end of one timeline's steps, then simply start the steps in the next section (which has already been clearly delineated by the border/header).

Reply:
My vague concern is there's not enough of a "real" boundary between these. The border and header are all supposed to be stylistic (they're defined in CSS), so without CSS it becomes just two `<ol>`s right next to each other. Also, steps are always going to know which timeline they're going to run on, so it kind of makes sense to write that way.

I'm actually thinking of removing the CSS headers to go with this change.

If I could come up with a clear and concise language for "issue the steps defined immediately following this list, on the timeline specified there" then I think it would work a bit better than just "Issue the subsequent steps".

Reply:
Maybe a better solution would just be to move the language back to the step it's issued from, and just say "Device timeline" at the top of device timeline steps.

Reply:
It will also be less redundant once we say something like "device.timeline" instead of "the device timeline of device"